### DISCUSSION TITLE
ASK: Why MN resources structure not found under IN CSE

### DISCUSSION THREAD

**Discussion by samuelbles07** [ankraft/ACME-oneM2M-CSE#107] [Q&A] [answered]
Hi, thank you for this awesome project!

I create IN in the same machine with MN, but different folder to seperate the database. After i create every resource under IN and MN, i see that IN cannot discover MN resources as pictures below

<img width="1440" alt="Screen Shot 2023-01-18 at 17 38 38" src="https://user-images.githubusercontent.com/18721002/213150433-8675dde3-5c44-4d4d-b89a-b56d0fec0f2e.png">

**WEBUI from IN**

<img width="1440" alt="Screen Shot 2023-01-18 at 17 38 42" src="https://user-images.githubusercontent.com/18721002/213150482-3499f179-f567-441a-bd24-74064cef1e78.png">

**WEBUI from MN**

I try to get CDoor latest instance, but request it from IN and the result as below

```
header = {
  "Accept": "application/json",
  "X-M2M-Origin": "in-cse/CAdmin",
   "X-M2M-RI": "hsj4sgfccit",
   "X-M2M-RVI": "3"
}

url = "http://localhost:8080/cse-mn/CDoor/control-lock"

response : '{"m2m:dbg": "resource not found"}'
```

debug in ACME-oneM2M-CSE IN folder

```
DEBUG   HTRE_24977 - ==> HTTP Request: cse-mn/CDoor/control-lock         HttpServer.py:205
DEBUG   HTRE_24977 - Operation: RETRIEVE                                 HttpServer.py:206
DEBUG   HTRE_24977 - Headers:                                            HttpServer.py:207
        Host: localhost:8080
        User-Agent: python-requests/2.25.1
        Accept-Encoding: gzip, deflate
        Accept: application/json
        Connection: keep-alive
        Content-Type: application/json;ty=4
        X-M2M-Origin: in-cse/CAdmin
        X-M2M-Ri: hsj4sgfccit
        X-M2M-Rvi: 3
DEBUG   HTRE_24977 - RETRIEVE ID: cse-mn/CDoor/control-lock,         RequestManager.py:180
        originator: in-cse/CAdmin
DEBUG   HTRE_24977 - Process RETRIEVE request for id:                     Dispatcher.py:81
        None|cse-mn/CDoor/control-lock
DEBUG   HTRE_24977 - Discover/Retrieve resources (rcn: attributes, fu:   Dispatcher.py:121
        conditionalRetrieval, drt: structured, fc: fu:
        conditionalRetrieval, fo: AND, rcn: attributes, attributes: {},
        sqi: None)
DEBUG   HTRE_24977 - Retrieve local resource: None|None for originator:  Dispatcher.py:281
        in-cse/CAdmin
DEBUG   HTRE_24977 - <== HTTP Response (notFound):                       HttpServer.py:617
        Headers: {'Server': 'ACME 0.11.1', 'X-M2M-RSC': '4004',
        'X-M2M-RI': 'hsj4sgfccit', 'X-M2M-RVI': '3', 'X-M2M-OT':
        '20230118T104003,667449', 'Content-Type': 'application/json'}
        Body: {'m2m:dbg': 'resource not found'}
```

But i see that MN CSE always successfully request keep alive periodically. I wonder what i do wrong? 

Thanks

**Comment by reinaortega** [comment]
Hi, 

Well, I can see that you are sending the request to the IN (8080), that should be 8081. And secondly, the Origin (From parameter) should be the id of the IN (I guess "/id-in", including the slash)

Cheers

**Comment by ankraft** [comment, marked as answer]
First of all thank you for your kind words. 
It looks like you did the main setup for the remote registration correctly.

Looking at your request I see that you want to do a transit request. You send your request to the IN-CSE (that is where also the MN-CSE has registered). You now expect that the IN-CSE forwards your request to the MN-CSE. This should be the case, however it seems that your target URL is missing the CSE-ID of the MN-CSE, ie. "/id-mn". Without that prefix the IN-CSE only targets resources that it hosts, and doesn't forward

Please try to modify your target URI to SP-Relative form: **/id-mn/cse-mn/CDoor/control-lock**. Don't forget: the http URL then will look like this: https:/\<address>/~/id-mn/cse-mn/CDoor/control-lock

**Reply by samuelbles07** [reply]
Yes exactly transit request! it works, thanks!
